### PR TITLE
refactor: [FC-0072] migrate project to swift 6

### PR DESCRIPTION
**Pull request by IvanStepanok**
This commit updates the entire project to Swift 6, incorporating language features like `Sendable` conformance and improved concurrency support. Key changes include:

- `Sendable` conformance added across the codebase for thread safety.
- Extensive migration to `async/await` for modern concurrency handling.
- New SwiftGen rule introduced to ensure compatibility with Swift 6.
- Significant updates to the test suite:
  - Most tests refactored and optimized for reliability and faster execution.
  - Enhanced compatibility with CI/CD pipelines.

BREAKING CHANGE: The project now requires Swift 6, and dependencies or integrations using earlier Swift versions may need adjustments to ensure compatibility.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @IvanStepanok!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/openedx-mobile-maintainers`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

**Review by volodymyr-chekyrta — DISMISSED** [review]

**Review by rnr — APPROVED** [review]

File: Course/Course/Presentation/Video/PlayerViewControllerHolder.swift [hunk 1]
```diff
@@ -5,10 +5,12 @@
 //  Created by Vadim Kuznetsov on 20.03.24.
 //
 
-import AVKit
-import Combine
+@preconcurrency import AVKit
+@preconcurrency import Combine
+import Core
```
[the remark this answers is not in the document]
Comment (IvanStepanok) [reply Course/Course/Presentation/Video/PlayerViewControllerHolder.swift:10]: fixed

File: Course/Course/Presentation/Video/VideoPlayerViewModel.swift [hunk 2]
```diff
@@ -137,7 +138,7 @@ public class VideoPlayerViewModel: ObservableObject {
     
     private func generateSelectedLanguage() {
         if let selectedLanguage = languages.first(where: {
-            $0.language == Locale.current.languageCode
+            $0.language == Locale.current.language.languageCode?.identifier
```
[the remark this answers is not in the document]
Comment (IvanStepanok) [reply Course/Course/Presentation/Video/VideoPlayerViewModel.swift:141]: 'languageCode' was deprecated in iOS 16: renamed to 'language.languageCode.identifier'

File: Discussion/Discussion/Presentation/DiscussionTopics/DiscussionSearchTopicsViewModel.swift [hunk 3]
```diff
@@ -160,15 +162,20 @@ public class DiscussionSearchTopicsViewModel<S: Scheduler>: ObservableObject {
...
-                    animated: true
-                )
-            }))
+                    self.router.showThread(
+                        thread: thread,
+                        postStateSubject: self.postStateSubject,
+                        isBlackedOut: false,
+                        animated: true
+                    )
+            })
```
[the remark this answers is not in the document]
Comment (IvanStepanok) [reply Discussion/Discussion/Presentation/DiscussionTopics/DiscussionSearchTopicsViewModel.swift:177]: done

File: OpenEdX/Data/DiscoveryPersistence.swift [hunk 4]
```diff
@@ -61,20 +61,19 @@ public class DiscoveryPersistence: DiscoveryPersistenceProtocol {
...
     }
     
     public func loadCourseDetails(courseID: String) async throws -> CourseDetails {
-        try await context.perform {[context] in
             let request = CDCourseDetails.fetchRequest()
             request.predicate = NSPredicate(format: "courseID = %@", courseID)
+        return try await container.performBackgroundTask { context in
             guard let courseDetails = try? context.fetch(request).first else { throw NoCachedDataError() }
             return CourseDetails(
                 courseID: courseDetails.courseID ?? "",
```
[the remark this answers is not in the document]
Comment (IvanStepanok) [reply OpenEdX/Data/DiscoveryPersistence.swift:79]: done

File: OpenEdX/Data/DiscoveryPersistence.swift [hunk 5]
```diff
@@ -94,9 +93,9 @@ public class DiscoveryPersistence: DiscoveryPersistenceProtocol {
         }
     }
     
-    public func saveCourseDetails(course: CourseDetails) {
-        context.perform {[context] in
-            let newCourseDetails = CDCourseDetails(context: self.context)
+    public func saveCourseDetails(course: CourseDetails) async {
+        await container.performBackgroundTask { context in
+        let newCourseDetails = CDCourseDetails(context: context)
```
[the remark this answers is not in the document]
Comment (IvanStepanok) [reply OpenEdX/Data/DiscoveryPersistence.swift:98]: done

File: OpenEdX/Managers/PipManager.swift [hunk 6]
```diff
@@ -90,7 +92,7 @@ public class PipManager: PipManagerProtocol {
         try await navigate(to: holder)
     }
     
-    public func pauseCurrentPipVideo() {
+    @MainActor public func pauseCurrentPipVideo() {
```
[the remark this answers is not in the document]
Comment (IvanStepanok) [reply OpenEdX/Managers/PipManager.swift:95]: fixed

File: Profile/Profile/Presentation/DatesAndCalendar/DatesAndCalendarViewModel.swift [hunk 7]
```diff
@@ -144,8 +145,9 @@ public class DatesAndCalendarViewModel: ObservableObject {
         updateCoursesCount()
     }
     
-    func clearAllData() {
-        calendarManager.clearAllData(removeCalendar: true)
+    @MainActor
```
[the remark this answers is not in the document]
Comment (IvanStepanok) [reply Profile/Profile/Presentation/DatesAndCalendar/DatesAndCalendarViewModel.swift:148]: fixed

File: Profile/Profile/Presentation/DatesAndCalendar/DatesAndCalendarViewModel.swift [hunk 8]
```diff
@@ -427,6 +424,7 @@ public class DatesAndCalendarViewModel: ObservableObject {
             }
     }
     
+    @MainActor
```
[the remark this answers is not in the document]
Comment (IvanStepanok) [reply Profile/Profile/Presentation/DatesAndCalendar/DatesAndCalendarViewModel.swift:427]: fixed

File: Profile/Profile/Presentation/Settings/SettingsViewModel.swift [hunk 9]
```diff
@@ -116,6 +121,7 @@ public class SettingsViewModel: ObservableObject {
             }.store(in: &cancellables)
     }
     
+    @MainActor
```
[the remark this answers is not in the document]
Comment (IvanStepanok) [reply Profile/Profile/Presentation/Settings/SettingsViewModel.swift:124]: fixed

**Comment by IvanStepanok** [comment]
@rnr Thanks for reviewing! All feedback are addressed

**Review by rnr — APPROVED** [review]

**Review by volodymyr-chekyrta — APPROVED** [review]

**Pull request merged**